### ISSUE TITLE
Completion of error handling

### ISSUE THREAD

**Issue by elfring** [bwall/pemcracker#1]
Would you like to add more error handling for return values from functions like the following?
- [fopen](http://pubs.opengroup.org/onlinepubs/9699919799/functions/fopen.html "Open a stream.") ⇒ [crackThread](https://github.com/bwall/pemcracker/blob/97e86e7e5013a94bab27c55cc13acea5e4a2411a/main.c#L50)
- [pthread_create](http://pubs.opengroup.org/onlinepubs/9699919799/functions/pthread_create.html "Thread creation") ⇒ [main](https://github.com/bwall/pemcracker/blob/97e86e7e5013a94bab27c55cc13acea5e4a2411a/main.c#L158)

**Comment by bwall** [comment]
Have you experienced any crashes as the result of this?

If not, I may create a "Code quality" milestone, which will handle all issues not directly related to current bugs.  I agree this and #2 should be resolved, but I have limited free time to work on this project currently.

**Comment by elfring** [comment]
I suggest to avoid [ignorance of return values](https://cwe.mitre.org/data/definitions/252.html "Unchecked return values") a bit more.
Would you like to detect every error situation as early as possible?

**Comment by bwall** [comment]
Yes, but at this moment, this and #2 appear to be code quality issues, not bugs which result in a security compromise or compromise of functionality.  I am aware of the quality of the code, and I am aware of ignorance of return values.  I have limited time allocated to this project, and even less now that I'm aware that JohnTheRipper supports encrypted PEMs.  

Please feel free to submit a PR with fixes.  If you have test cases which show either disruptions in functionality/stability/security, I will be happy to resolve them.
